hs/src/Reach/Connector/ALGO.hs

```diff
@@ -2028,10 +2028,23 @@ ce = \case
         incResource R_App ro
         makeTxn1 "ApplicationID"
         let as' = (DLA_Literal $ DLL_Int at $ fromIntegral $ sigStrToInt sig) : as
-        forM_ as' $ \a -> do
-          ca a
-          ctobs $ argTypeOf a
-          makeTxn1 "ApplicationArgs"
+        let processArg a = do
+              ca a
+              ctobs $ argTypeOf a
+              makeTxn1 "ApplicationArgs"
+        let processArgTuple tas = do
+              cconcatbs $ map (\a -> (argTypeOf a, ca a)) tas
+              makeTxn1 "ApplicationArgs"
+        -- If there are more than 15 arguments, args 15+ must be packed as a tuple
+        -- into argument at index 15.  Note that the API id is arg 0, so we have
+        -- a total of up to 16 arguments.
+        case 16 < (length as') of
+          False -> do
+            forM_ as' processArg
+          True -> do
+            let (as'15, as'More) = splitAt 15 as'
```

Comment:
Why does this split at 15 but the others split at 14? Because it includes the method number? I believe the conflict you have with ALGO.hs is because I changed this to not have the number

Reply:
Yes, because of the method number.  I guess maybe I should rebase this then to solve the conflicts.  But maybe I'll wait until I finish figuring out the solidity stuff.